src/main/java/uk/gov/hmcts/reform/roleassignment/domain/service/createroles/CreateRoleAssignmentOrchestrator.java

```diff
@@ -49,11 +49,11 @@ public void addExistingRoleAssignments(RoleAssignmentRequest roleAssignmentReque
         }
     }
 
-    public void updateRequestStatus(RoleAssignmentRequest roleAssignmentRequest) {
-        roleAssignmentRequest.roleRequest.status = Status.APPROVED;
-        for (RequestedRole requestedRole : roleAssignmentRequest.requestedRoles) {
+    public void updateRequestStatus(AssignmentRequest assignmentRequest) {
+        assignmentRequest.request.status = Status.APPROVED;
+        for (RequestedRole requestedRole : assignmentRequest.requestedRoles) {
             if (!requestedRole.isApproved()) {
-                roleAssignmentRequest.roleRequest.status = Status.REJECTED;
+                assignmentRequest.request.status = Status.REJECTED;
             }
         }
```

Comment:
we need a reverse logic to make all requestedRole as rejected if request is rejected. this requires some logging about which role got rejected.